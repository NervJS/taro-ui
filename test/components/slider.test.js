import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtSlider from '../../.temp/components/slider/index'

describe('AtSlider Snap', () => {
  it('render initial AtSlider', () => {
    const component = renderToString(<AtSlider />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props value', () => {
    const component = renderToString(<AtSlider value={50} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props step', () => {
    const component = renderToString(<AtSlider step={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props min', () => {
    const component = renderToString(<AtSlider min={50} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props max', () => {
    const component = renderToString(<AtSlider max={200} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props disabled', () => {
    const component = renderToString(<AtSlider disabled />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props activeColor', () => {
    const component = renderToString(<AtSlider activeColor='#4285F4' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props backgroundColor', () => {
    const component = renderToString(<AtSlider backgroundColor='#BDBDBD' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props blockColor', () => {
    const component = renderToString(<AtSlider blockColor='#4285F4' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props blockSize', () => {
    const component = renderToString(<AtSlider blockSize={24} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtSlider -- props showValue', () => {
    const component = renderToString(<AtSlider showValue />)
    expect(component).toMatchSnapshot()
  })
})
