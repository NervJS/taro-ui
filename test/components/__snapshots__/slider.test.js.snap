// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSlider Snap render AtSlider -- props activeColor 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#4285F4;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props backgroundColor 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#BDBDBD;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props blockColor 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#4285F4;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props blockSize 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:24px;height:24px;background-color:#ffffff;margin-top:-12px;margin-left:-12px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props disabled 1`] = `"<div class=\\"at-slider at-slider--disabled\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props max 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props min 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:100%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:100%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"50\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props showValue 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div><div class=\\"at-slider__text\\">0</div></div>"`;

exports[`AtSlider Snap render AtSlider -- props step 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render AtSlider -- props value 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:50%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:50%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"50\\"/></div></div>undefined</div></div>undefined</div>"`;

exports[`AtSlider Snap render initial AtSlider 1`] = `"<div class=\\"at-slider\\"><div class=\\"at-slider__inner\\"><div class=\\"weui-slider-box\\"><div class=\\"weui-slider\\"><div class=\\"weui-slider__inner\\" style=\\"background-color:#e9e9e9;\\"><div style=\\"width:0%;background-color:#6190e8;\\" class=\\"weui-slider__track\\"></div><div style=\\"left:0%;width:28px;height:28px;background-color:#ffffff;margin-top:-14px;margin-left:-14px;\\" class=\\"weui-slider__handler\\"></div><input type=\\"hidden\\" name=\\"\\" value=\\"0\\"/></div></div>undefined</div></div>undefined</div>"`;
